feat(curriculum): add Artifact Provenance Auditor workshop (#68860)
freeCodeCamp/freeCodeCamp: curriculum/challenges/english/blocks/workshop-artifact-provenance-auditor/6a57c331ae94fa4631b60872.md

Reviewer: I would suggest also adding a new object to the list to use in the tests, so you can check that `buildSummary` really do what you are asking to do, and that `buildSummary(457)` with the new object with `id` (for example) `457` is correctly summarised.

---
id: 6a57c331ae94fa4631b60872
title: Step 19
challengeType: 1
dashedName: step-19
---

# --description--

> Now return a template literal summarizing the artifact. Put the title on the first line, then one line each for the category, curator name, current gallery, and display status. Use `currentLocation.gallery` for the gallery.

Reviewer 2: ```suggestion
Next, return a template literal summarizing the artifact. Put the title on the first line, followed by one line each for the category, curator's name, current gallery, and display status. Use `currentLocation.gallery` for the gallery.
```

> For example, `buildSummary(101)` returns:

Reviewer 2: ```suggestion
For example, if you call `buildSummary(101)`, it should return:
```

  if (artifact && !artifact.tags.includes(tag)) {
    artifact.tags.push(tag);
  }
}

console.log("UPDATED TAGS");
addTag(101, "royal");
console.log(collection[101].tags);

function moveArtifact(id, gallery, year) {
  const artifact = collection[id];
  if (artifact) {
    artifact.locations.push({ gallery, year });
  }
}

console.log("UPDATED LOCATIONS");
moveArtifact(102, "Hall B", 2026);
console.log(collection[102].locations);

function toggleDisplayStatus(id) {
  const artifact = collection[id];
  if (artifact) {
    artifact.onDisplay = !artifact.onDisplay;
  }
}

console.log("DISPLAY STATUS");
console.log(collection[102].onDisplay);
toggleDisplayStatus(102);
console.log(collection[102].onDisplay);

function updateCurator(id, name) {
  const artifact = collection[id];
  if (artifact) {
    artifact.curator.name = name;
  }
}

console.log("UPDATED CURATOR");
updateCurator(101, "Fran Sinclair");
console.log(collection[101].curator.name);

function buildSummary(id) {
  const artifact = collection[id];
  if (!artifact) {
    return "Artifact not found";
  }
  const currentLocation = artifact.locations[artifact.locations.length - 1];
--fcc-editable-region--

> 

Reviewer 2: Indent the empty line b/w the markers with 2 spaces.
